syntax = "proto2";

package rokid.open;

enum SpeechErrorCode {
  SUCCESS = 0;
  AUTH_FAILED = 1;
  UNAUTHENTICATED = 2;
  CONNECTION_EXCEED = 3;
  RESOURCE_EXHASTED = 4;
  BUSY = 5;
  INTERNAL = 6;
}

enum ReqType {
  START = 0;
  VOICE = 1;
  END = 2;
  // used for SpeechRequest
  TEXT = 3;
}

message AuthRequest {
  required string key = 1;
  required string device_type_id = 2;
  required string device_id = 3;
  required string service = 4;
  required string version = 5;
  required string timestamp = 6;
  required string sign = 7;
}

message AuthResponse {
  // 0: success
  // -1: failed
  required SpeechErrorCode result = 1;
}

message AsrRequest {
  required int32 id = 1;
  // 0: start
  // 1: voice data
  // 2: end
  required ReqType type = 2;
  optional bytes voice = 3;
  optional string lang = 4;
  optional string codec = 5;
  // vt = voice trigger
  optional string vt = 6;
}

message AsrResponse {
  required int32 id = 1;
  required SpeechErrorCode result = 2;
  optional string asr = 3;
  optional bool finish = 4;
}

message TtsRequest {
  required int32 id = 1;
  required string text = 2;
  optional string declaimer = 3;
  optional string codec = 4;
}

message TtsResponse {
  required int32 id = 1;
  required SpeechErrorCode result = 2;
  optional string text = 3;
  optional bytes voice = 4;
  optional bool finish = 5;
}

message SpeechRequest {
  required int32 id = 1;

  // 0: start
  // 1: voice data
  // 2: end
  // 3: text
  required ReqType type = 2;

  optional bytes voice = 3;

  optional string asr = 4;

  // zh
  // en
  optional string lang = 5;

  // pcm
  // opu
  // opu2
  optional string codec = 6;

  // vt = voice trigger
  // deprecated
  optional string vt = 7;

  // json string: options of framework
  // {
  //   "voice_trigger": "xxx",
  //   "voice_power": "xxx",
  //   "trigger_start": "xxx",
  //   "trigger_end": "xxx",
  //   "stack": "xxx"
  // }
  optional string framework_options = 8;

  // json string: options of skills
  // {
  //   "media_player_status": "xxx"
  // }
  optional string skill_options = 9;
}

message SpeechResponse {
  required int32 id = 1;

  required SpeechErrorCode result = 2;

  optional string asr = 3;

  optional string nlp = 4;

  optional string action = 5;

  optional bool finish = 6;

  // {
  //   "activation": "fake|reject|accept|none"
  // }
  optional string extra = 7;
}

